### PR TITLE
Add support for `noop` command

### PR DESCRIPTION
**Pull request by CraigFe**
... and refactor some of the tests to use Alcotest. Currently requires the dev version of Alcotest, which I will release shortly.

### CODE REVIEW

File: lib/uring/uring.ml [hunk 1]
```diff
@@ -65,9 +66,12 @@ type 'a t = {
   mutable dirty: bool; (* has outstanding requests that need to be submitted *)
 }
 
+let invalid_argf fmt = Format.kasprintf invalid_arg fmt
```
Comment (talex5) [inline lib/uring/uring.ml:69]: We already depend on `fmt`, so might as well use `Fmt.invalid_arg` instead.

File: tests/main.ml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+module Test_data = struct
+  let path = "output_file.txt"
+
+  let setup () =
+    let oc = open_out path in
+    output_string oc "A test file";
+    close_out oc
+
+  let get_fd () = Unix.openfile path [ O_RDONLY ] 0
```
Comment (talex5) [inline tests/main.ml:13]: Maybe `with_fd`, to avoid leaking it?

**Review by talex5 — COMMENTED** [review]
Thanks for cleaning this up!

File: tests/main.ml (hunk 2 above)
Comment (CraigFe) [reply tests/main.ml]: Good idea; done.

**Comment by CraigFe** [comment]
The CI is now happy with this. I'll make some follow-up PRs shortly.

**Pull request merged**